ferminet/networks.py

```diff
@@ -440,17 +440,30 @@ def logdet_matmul(xs: Sequence[jnp.ndarray],
     determinant (or product of the i-th determinant in each spin channel, if
     full_det is not used).
   """
-  slogdets = [slogdet(x) for x in xs]
-  sign_in, logdet = functools.reduce(
-      lambda a, b: (a[0]*b[0], a[1]+b[1]), slogdets)
+  # Special case if there is only one electron in any channel
```

Comment:
If there's any spin channel with more than one electrons, then the slogdets is not empty and so the if branch is taken. I think this is fine (I have run the lithium atom for hundreds of thousands of iterations as a test) but the behaviour doesn't match this comment.

How about moving the evaluation of dets into the else branch, as that's the only place its used?

Reply:
If `full_det=False`, there may be a channel with a single electron and one with two (e.g., in Lithium). So, we want to compute the determinant of the channel with one electron and the log-det for the two-electron channel. So, the result may also be used in the if branch.
In line 458, the `dets` variable is combined with the log determinants.